domains/zeto/pkg/types/states.go

```diff
@@ -94,6 +94,38 @@ var ZetoNFTokenABI = &abi.Parameter{
 	},
 }
 
+type TransactionData struct {
+	Salt *pldtypes.HexUint256 `json:"salt"`
+	Data pldtypes.HexBytes    `json:"data"`
+	hash *pldtypes.HexUint256
+}
+
+var TransactionDataABI = &abi.Parameter{
+	Name:         "TransactionData",
+	Type:         "tuple",
+	InternalType: "struct TransactionData",
+	Components: abi.ParameterArray{
+		{Name: "salt", Type: "bytes32"},
+		{Name: "data", Type: "bytes"},
+	},
+}
+
+// TODO: is this an appropriate hashing method for this state data?
```

Comment:
Open question here. Zeto has chosen to provide its own state hashes, so it must hash the transaction metadata as well. I put in a simple SHA-256 hash, but I'm not sure this is actually a correct choice.

Reply:
Why would the same algorithm not be used?

I assume for this PR to be complete (and consistent with Noto), the Zeto transaction proofs must include validation of the hash?

Reply:
I don't think Poseidon is great for arbitrary sequences of bytes. It's more for fixed length inputs, and chosen for ease of use in ZKPs (which I don't think is applicable for this piece of data).

Reply:
I am implying that the data is passed "in the clear" on chain and could be validated by the receiver, but is not obscured or validated by a ZKP.

Reply:
Note discovered that we didn't update `ValidateStateHashes()` in back and forth in testing - adding that as part of validating your change.

Reply:
> I am implying that the data is passed "in the clear" on chain and could be validated by the receiver, but is not obscured or validated by a ZKP.

I think you mean the salted-hash of the data (no not leaking anything) as the data itself is passed private.

Make sense as I think about it that no ZKP is needed for this, as there is no processing of the data to validate (such as total conservation of value etc.)

Reply:
Thanks for flagging (and fixing) the validation of the state hashes.

And yes you're completely right on what's passed on chain: it's just the data hashes, not the full contents of the data - and it's passed to the smart contract only, but not incorporated into any proof.